### PR TITLE
added chatbot functionality, started modularization

### PR DESCRIPTION
**Pull request by naterchrdsn**

### CODE REVIEW

**Comment by martindale** [comment]
Nice work!  At almost 2,000 lines to review, would you mind switching to 2 spaces instead of 4?  Also, I think maybe the data stuff you're adding (the various "facts") could be broken out into standalone modules — that could be a quick PR to the existing bot, too.

**Comment by naterchrdsn** [comment]
no problem

**Comment by naterchrdsn** [comment]
still have to finish slack support...

TODO:
- reformat message from plugins for slack api
- fix greeting new team members on slack
- load discord-specific plugins (moderation and musicplayer)
- break out plugins in random plugin to be separate modules? - might happen later

File: plugins/cocktail-lookup.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+const Doorman = require('../../doorman');
```
Comment (martindale) [inline plugins/cocktail-lookup.js]: Let's break these out into submodules, with separate repos.  Will keep the codebase smaller and easier to understand for contributors.

Comment (naterchrdsn) [inline plugins/cocktail-lookup.js]: That's exactly what I've done with my bot, [Kirbi](https://github.com/Richardson-Media-House/Kirbi)

Comment (naterchrdsn) [inline plugins/cocktail-lookup.js]: in fact, the modules are the exact same as with that bot, so we could just import those via the package.json and config as I did with Kirbi

that cool?

**Pull request closed without merging**